### PR TITLE
Oauth

### PR DESCRIPTION
**Pull request by GAUTAMRAJU15**
pls review

### CODE REVIEW

File: src/components/Affiliate/AffiliateRegister.js [hunk 1]
```diff
@@ -37,12 +47,32 @@ export default class AffiliateRegister extends Component {
...
-  
+
   handleNameChange = (event) => {
     const { name, value } = event.target;
     this.setState({ [name]: value });
   }
 
+  handleSubmit(evt) {
+    if (!this.canBeSubmitted()) {
+      evt.preventDefault();
```
Comment (ShankyRana) [inline src/components/Affiliate/AffiliateRegister.js]: use this one time only

File: src/components/Affiliate/AffiliateRegister.js [hunk 2]
```diff
@@ -102,3 +132,13 @@ export default class AffiliateRegister extends Component {
     );
   }
 }
+
+const mapStateToProps = state => ({
```
Comment (ShankyRana) [inline src/components/Affiliate/AffiliateRegister.js]: handle error correctly add a new duck for affiliate error.

File: src/sagas/auth.js [hunk 3]
```diff
@@ -111,6 +111,20 @@ export function* fetchRoles() {
   }
 }
 
+export function* affiliateSuccess(action) {
+  try {
+    yield put(load());
+    const res = yield call(api.GET, 'user/sendmail/affiliate', action.data);
```
Comment (ShankyRana) [inline src/sagas/auth.js]: you have to send query.
like the example i gave you

Comment (ShankyRana) [reply src/sagas/auth.js]: action.data is not required, its a get request

File: src/components/Affiliate/AffiliateRegister.js (hunk 2 above)
Comment (Devilla) [reply src/components/Affiliate/AffiliateRegister.js]: okay

**Pull request merged**